spec.bs

```diff
@@ -2091,6 +2162,402 @@ a {{ReportingBrowserSignals}} |browserSignals|, and a [=direct from seller signa
      :: |reportingMacroMap|
 </div>
 
+# Additional Bids and Negative Targeting # {#additional-bids-and-negative-targeting}
+
+## createAuctionNonce() ## {#create-auction-nonce}
+
+*This first introductory paragraph is non-normative.*
+
+{{Window/navigator}}.{{Navigator/createAuctionNonce()}} creates an auction nonce, a one-time
+[=version 4 UUID=] uniquely associated with a single call to
+{{Window/navigator}}.{{Navigator/runAdAuction()}}. For multi-seller auctions, this ID is uniquely
+associated with all {{AuctionAdConfig/componentAuctions}}.
+This nonce will need to be passed back in via a subsequent call to
+{{Window/navigator}}.{{Navigator/runAdAuction()}} via the {{AuctionAdConfig}}. This is currently
+only needed for auctions that use additional bids, for which the auction nonce will be included in
+each additional bid as a way of ensuring that those bids are only used in the auctions for which
+they were intended.
+
+<xmp class="idl">
+[SecureContext]
+partial interface Navigator {
+  Promise<DOMString> createAuctionNonce();
+};
+</xmp>
+
+<div algorithm="createAuctionNonce()">
+The <dfn for=Navigator method>createAuctionNonce()</dfn> method steps are:
+
+  1. Let |p| be [=a new promise=].
+  1. Run the following steps [=in parallel=]:
+    1. Let |nonce| be the [=string representation=] of a [=version 4 UUID=].
+
+    <div class=note>
+    Because we're going in parallel:
+      * There is no guarantee that the promise will be resolved before other tasks get queued on the
+        main thread;
+      * ...which gives browsers the freedom to generate this UUID in another process, and
+        asynchronously send it back to the main thread at an arbitrary future time.
+    </div>
+    1. [=Queue a task=] to [=resolve=] |p| with |nonce|.

```

Comment:
You can't just queue a task. You need to specify a global. Please pull the global out before going in parallel, and then copy this pattern: https://html.spec.whatwg.org/C#unhandled-promise-rejections:queue-a-global-task.